lib/chef/provider/remote_file/network_file.rb

```diff
@@ -35,13 +42,23 @@ def initialize(source, new_resource, current_resource)
         # Fetches the file on a network share, returning a Tempfile-like File handle
         # windows only
         def fetch
-          tempfile = Chef::FileContentManagement::Tempfile.new(new_resource).tempfile
-          Chef::Log.debug("#{new_resource} staging #{@source} to #{tempfile.path}")
-          FileUtils.cp(@source, tempfile.path)
-          tempfile.close if tempfile
+          validate_identity(new_resource.remote_user, new_resource.remote_user_password, new_resource.remote_user_domain)
```

Comment:
`with_user_context` already calls `validate_identity` so this seems unnecessary here.

Also, it seems like the `UserIdentity` and `UserContext` mixins could be merged if the only method providers need to call is `with_user_context`